### PR TITLE
Issue 1120

### PR DESCRIPTION
**Pull request by lxning**
## Description

Please include a summary of the feature or issue being fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)
#1120 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the tests [UT/IT] that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced. 
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- UT/IT execution results

- Logs

## Checklist:

- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] New and existing unit tests pass locally with these changes?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -284,6 +284,7 @@ the backend workers convert "Bytearray to utf-8 string" when the Content-Type of
 * `model_server_home` : Torchserve home directory. 
 * `max_request_size` : The maximum allowable request size that the Torchserve accepts, in bytes. Default: 6553500
 * `max_response_size` : The maximum allowable response size that the Torchserve sends, in bytes. Default: 6553500
+* `limit_max_image_pixels` : Default value is true. If this is set to "false", PIL.Image.MAX_IMAGE_PIXELS = None in backend default vision handler.
```
Comment (HamidShojanazeri) [inline docs/configuration.md:287]: @lxning should we add some explanations that this setting would be helpful is running inference for images with size beyond 5MB (which I think is our current limit?).

File: docs/configuration.md [hunk 2]
```diff
@@ -284,6 +284,7 @@ the backend workers convert "Bytearray to utf-8 string" when the Content-Type of
 * `model_server_home` : Torchserve home directory. 
 * `max_request_size` : The maximum allowable request size that the Torchserve accepts, in bytes. Default: 6553500
 * `max_response_size` : The maximum allowable response size that the Torchserve sends, in bytes. Default: 6553500
+* `limit_max_image_pixels` : Default value is true (Use default [PIL.Image.MAX_IMAGE_PIXELS](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.MAX_IMAGE_PIXELS)). If this is set to "false", set PIL.Image.MAX_IMAGE_PIXELS = None in backend default vision handler for large image payload.
```
Comment (nskool) [inline docs/configuration.md:287]: Hi @lxning, could you add more details on how the `limit_max_image_pixels`, fixes the related issue #1120 ? Or, at least aids in the resolution of the issue? [Request to add details in the PR description]. Is the idea that a user should be able to limit the size of the input image?

File: ts/context.py [hunk 3]
```diff
@@ -11,19 +11,21 @@ class Context(object):
...
             "batch_size": batch_size,
             "server_name": "MMS",
-            "server_version": mms_version
+            "server_version": mms_version,
+            "limit_max_image_pixels": limit_max_image_pixels,
         }
         self.request_ids = None
         self.request_processor = None
         self._metrics = None
+        self._limit_max_image_pixels = True
```
Comment (nskool) [inline ts/context.py:28]: Could you elaborate on how this is further used

File: ts/torch_handler/vision_handler.py [hunk 4]
```diff
@@ -20,6 +20,9 @@ def initialize(self, context):
         super().initialize(context)
         self.ig = IntegratedGradients(self.model)
         self.initialized = True
+        properties = context.system_properties
+        if not properties.get("limit_max_image_pixels") :
+            Image.MAX_IMAGE_PIXELS = None
```
Comment (nskool) [inline ts/torch_handler/vision_handler.py:25]: Here, we're setting value to None, however, it does not seem to be clear what happens if this value is not None, does PILLOW automatically pick it up from context?

**Review by nskool — APPROVED** [review]

File: docs/configuration.md (hunk 2 above)
Comment (lxning) [reply docs/configuration.md:287]: the detailed information is provided in the above link

File: ts/torch_handler/vision_handler.py (hunk 4 above)
Comment (lxning) [reply ts/torch_handler/vision_handler.py:25]: Same as the above, the detailed information is provided in pillow link

**Pull request merged**